src/k8s/pkg/k8sd/setup/kube_proxy.go

```
@@ -9,14 +10,26 @@ import (
 )
 
 // KubeProxy configures kube-proxy on the local node.
-func KubeProxy(snap snap.Snap, hostname string, podCIDR string) error {
-	if _, err := snaputil.UpdateServiceArguments(snap, "kube-proxy", map[string]string{
+func KubeProxy(ctx context.Context, snap snap.Snap, hostname string, podCIDR string) error {
+	serviceArgs := map[string]string{
 		"--cluster-cidr":         podCIDR,
 		"--healthz-bind-address": "127.0.0.1",
 		"--hostname-override":    hostname,
 		"--kubeconfig":           path.Join(snap.KubernetesConfigDir(), "proxy.conf"),
 		"--profiling":            "false",
-	}, nil); err != nil {
+	}
+
+	onLXD, err := snap.OnLXD(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to check if on lxd: %w", err)
+	}
```

same here, let's eat the error or log something